### PR TITLE
chore: [CDS-85482]: add support for execution labels in pipeline studio

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

Jira: https://harness.atlassian.net/browse/CDS-88342

- This PR adds support for advanced configuration for the pipeline studio
- Onboards  Execution Labels for pipelines
- Updates Tags popover to a light theme in pipeline listing pages
- Combines Execution labels and tags in the pipeline listing and execution pages for the v1 pipeline

https://github.com/harness/harness-core-ui/assets/100121280/e0e35135-ddf4-4c2f-834c-959048b39035

image
image
image
image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by PratyushGar** [comment]
trigger messagecheck

File: src/modules/70-pipeline/y1/components/AdvancedOptions/AdvancedOptionsY1.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+      set(data, 'labels', executionLabels)
+      onApplyChanges(omit(data, 'publicAccessResponse'))
+    },
+    [onApplyChanges]
+  )
+
+  const executionLabels = Object.keys(defaultTo(pipeline?.labels, {}))?.map((key: string) => {
+    const value = pipeline?.labels?.[key]
+    return {
+      id: uuid('', nameSpace()),
```
Comment (rameshx) [inline src/modules/70-pipeline/y1/components/AdvancedOptions/AdvancedOptionsY1.tsx:105]: `uuid` will be called on every render, but will not be used as `enableReinitialize` on Formik is not true. Can we nest this in `useMemo`?

File: src/modules/70-pipeline/y1/components/RightBar/RightBarY1.tsx [hunk 2]
```diff
@@ -42,6 +49,28 @@ export function RightBarY1(): JSX.Element {
...
+        variation={ButtonVariation.TERTIARY}
+        onClick={() => {
+          updatePipelineView({
+            ...pipelineView,
+            isDrawerOpened: true,
+            drawerData: { type: DrawerTypesY1.AdvancedOptions },
+            isSplitViewOpen: false,
+            splitViewData: {}
+          })
+        }}
```
Comment (rameshx) [inline src/modules/70-pipeline/y1/components/RightBar/RightBarY1.tsx:65]: We can add a common handler for both runtime and advanced options `onClick` and pass `DrawerTypesY1.AdvancedOptions` or `DrawerTypesY1.RuntimeInputs`.

File: src/modules/70-pipeline/y1/components/AdvancedOptions/AdvancedOptionsY1.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/y1/components/AdvancedOptions/AdvancedOptionsY1.tsx]: Yes, will do

File: src/modules/70-pipeline/y1/components/RightBar/RightBarY1.tsx (hunk 2 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/y1/components/RightBar/RightBarY1.tsx]: Sure

**Review by rameshx — DISMISSED** [review]

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**